### PR TITLE
Simplified view

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
This PR adds extra authoring options / url switches to do the following:

- showUserInterface: true/false
- earthquakesDisplayAllOnStart: true/false (same functionality as dragging slider to the right)
- sizeToFitBounds: true/false - standard behavior is to use four corners for a bounding box, but setting this to false will use a center point and a zoom level that can be specified in authoring.
- centerLat, centerLng: coordinates for the center
- centeredInitialZoom: 6 (or another number) sets the initial zoom when using center point mode.
- allowDrag: true / false (default is true, if set to false the user cannot pan the map)
- zoomMax and zoomMin: fairly self-explanatory, if set to -1 they are ignored. They will not allow a zoom level greater than the max supported by the underlying map tiles.

I also updated as many dependencies as possible to fix some NPM warnings.

[#171616026]

### CODE REVIEW

File: js/components/seismic-eruptions-map.js [hunk 1]
```diff
@@ -93,7 +100,11 @@ export default class SeismicEruptionsMap extends PureComponent {
       // TOOD: remove, it can be useful to tweak position of plate arrows.
       console.log('lat:', e.latlng.lat, 'lng:', e.latlng.lng)
     })
-
+    if (!config.sizeToFitBounds) {
+      // if we are not using bounds, and instead using a center point, force a refresh of the viewport to
+      // get the earthquakes to render on first load
+      setTimeout(() => this.handleMapViewportChanged(), 400)
```
Comment (pjanik) [inline js/components/seismic-eruptions-map.js:106]: This seems a bit random. Why timeout and why 400ms?

File: js/components/seismic-eruptions-map.js [hunk 2]
```diff
@@ -39,14 +41,19 @@ export default class SeismicEruptionsMap extends PureComponent {
     super(props)
     this.state = {
       selectedEarthquake: null,
-      selectedVolcano: null
+      selectedVolcano: null,
+      scaleWidth: 0,
+      scaleHeight: 0,
+      zoom: 0,
+      mapViewport: { center: { lat: config.centerLat, lon: config.centerLng }, zoom: config.centeredInitialZoom }
```
Comment (pjanik) [inline js/components/seismic-eruptions-map.js:48]: Do we need zoom included in the state twice?

File: js/components/seismic-eruptions-map.js [hunk 3]
```diff
@@ -164,6 +185,21 @@ export default class SeismicEruptionsMap extends PureComponent {
...
+      // ensure map remains with the correct center.
+      const { mark2DViewModified } = this.props
+      const nextViewport = this.refs.map.viewport
+      // using the + and - buttons will not change the center, only double-click and mouse wheel
+      if (nextViewport.center !== INITIAL_CENTER) {
+        nextViewport.center = INITIAL_CENTER
+        this.setState({ mapViewport: nextViewport })
+        mark2DViewModified(false)
+      }
+    }
```
Comment (pjanik) [inline js/components/seismic-eruptions-map.js:200]: I'm not sure if I get the idea here. Why do we need to read the new viewport and overwrite center point?

File: js/components/seismic-eruptions-map.js (hunk 2 above)
Comment (dstrawberrygirl) [reply js/components/seismic-eruptions-map.js:48]: You're right, I don't think we need to be setting state. I'll adjust the PR accordingly, thanks!

File: js/components/seismic-eruptions-map.js (hunk 3 above)
Comment (dstrawberrygirl) [reply js/components/seismic-eruptions-map.js:200]: If you scroll in and out with your mouse, your center point will become the area where your mouse is, enabling you to traverse the map without panning it, but just using the zoom controls. There was a method in leaflet that should have prevented this (`setZoomAround`) but I was unable to get it to work in this context, so I re-set the center after a zoom instead as a workaround.

File: js/components/seismic-eruptions-map.js (hunk 1 above)
Comment (dstrawberrygirl) [reply js/components/seismic-eruptions-map.js:106]: copy/paste from elsewhere - good catch!

File: js/components/seismic-eruptions-map.js (hunk 3 above)
Comment (pjanik) [reply js/components/seismic-eruptions-map.js:200]: This Leaflet wrapper isn't perfect, right. But if there is a Leaflet option, we can definitely use it by accessing Leaflet instance directly. It's even done in this file:
https://github.com/concord-consortium/seismic-explorer/blob/17b8c12488ee8d494b3572f73b2c2ceb4f038eb2/js/components/seismic-eruptions-map.js#L120

**Comment by dstrawberrygirl** [comment]
@pjanik I've reworked this to be as light as possible, but it does feel a lot less responsive on my machine than master. I ran a profiler and saw spikes in the rendering in the canvas-layer when `requestAnimationFrame` was called but I'm not sure exactly where the issue is coming from. I am guessing it's related to the dependency update? I'm not seeing how my code (which is mostly only running if the user specified a center point) is running so much slowly without any url parameters compared to the production version.

**Comment by dstrawberrygirl** [comment]
Update: found a potential slowdown with the zoom level being updated each time the viewport changed. Have now removed that - still experiencing a slow down locally, still investigating

**Pull request merged**